scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,401 @@
+    @staticmethod
+    def check_if_service_is_active(service_name: str):
+        """
+        service_name for example : pacemaker
+        """
+        command = SYSTEMCTL_IS_ACTIVE_COMMAND.format(service_name)
+        _, stdout = StandbyNodeHealthChecker._run_command(command)
+        if stdout == "active":
+            print(f"Service {service_name} is active")
+            return True
+        print(f"Service {service_name} is not active")
+        return False
+
+    def get_ufm_ha_cluster_status_string(self):
+        command = UFM_HA_CLUSTER_COMMAND.format("status")
+        ret_code, res_string = StandbyNodeHealthChecker._run_command(command)
+        if ret_code != 0:
+            print("Failed to get ha cluster status")
+            res_string = ""
+        self._ufm_ha_status_string = res_string
+
+    def _get_status_value_from_ha_cluster_status(self, keyword: str):
+        for line in self._ufm_ha_status_string.splitlines():
+            line = line.strip()
+            if line.startswith(keyword):
+                return line.split(":", 1)[-1].strip().lower()
+        return None
+
+    def check_drbd_connectivity(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drbd_connectivity_status = self._get_status_value_from_ha_cluster_status(
+            "DRBD_CONNECTIVITY"
+        )
+        if drbd_connectivity_status != "[connected]":
+            print(
+                f"DRBD CONNECTIVITY status is {drbd_connectivity_status} and not Connected"
+            )
+            return False
+        print("DRBD CONNECTIVITY status is ok (Connected)")
+        return True
+
+    def check_drdb_resource(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drdb_resource_status = self._get_status_value_from_ha_cluster_status(
+            "DRBD_RESOURCE"
+        )
+        if drdb_resource_status != "[ha_data]":
+            print(f"DRDB RESOURCE status is {drdb_resource_status} and not ha_data")
+            return False
+        print("DRDB RESOURCE status is ok (ha_data)")
+        return True
+
+    def check_drbd_disk_state(self):
+        if not self._ufm_ha_status_string:
+            return False
+        drbd_disk_state = self._get_status_value_from_ha_cluster_status("DISK_STATE")
+        if drbd_disk_state != "[uptodate]":
+            print(f"DRBD DISK STATE status is {drbd_disk_state} and not UpToDate")
+            return False
+        print("DRBD DISK STATE status is ok (UpToDate)")
+        return True
+
+    def run_advanced_checks(self):
+        result = True
+        checks = [
+            StandbyNodeHealthChecker._check_pcs_status,
+            StandbyNodeHealthChecker._check_corosync_rings_status,
+            self.check_drdb_resource,
+            self.check_drbd_connectivity,
+            self.check_drbd_disk_state,
+        ]
+        for check in checks:
+            if not check():
+                result = False
+        check_with_arg = [
```

Comment:
you can use `partial` and unify the two checks

Reply:
I prefer to keep as it is for now, I think it is more clear what and how is being tested